xmtp_mls/src/client.rs

```diff
@@ -94,6 +94,25 @@ where
         Ok(group)
     }
 
+    pub fn find_groups(
```

Comment:
Can we put this directly on the `Group` struct, with `&Client` as an argument? Same with `create_group()`. Would prefer not to have every DB-related method replicated on the `Client` class

Reply:
I think in the FFI world it's going to make life easier to have all the group administration (`list`, `create`) live on the Client that we return to the native SDK. That way they don't have to deal with a bunch of different foreign object types and passing around references.

We could still do that mapping in the bindings, but it's messier. How about we leave it for now and we can always move things around later if we decide that it isn't as ergonomic?